lib/AST/ASTScopeCreation.cpp

```diff
@@ -0,0 +1,1061 @@
+  void visitPrefixOperatorDecl(       PrefixOperatorDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitPostfixOperatorDecl(     PostfixOperatorDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitGenericTypeParamDecl(   GenericTypeParamDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitAssociatedTypeDecl(       AssociatedTypeDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitModuleDecl(                       ModuleDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitParamDecl(                         ParamDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitEnumElementDecl(             EnumElementDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitIfConfigDecl(                   IfConfigDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitPoundDiagnosticDecl(     PoundDiagnosticDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+  void visitMissingMemberDecl(         MissingMemberDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+
+  // This declaration is handled from the PatternBindingDecl
+  void visitVarDecl(                VarDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+                          
+  // This declaration is handled from addChildrenForAllExplicitAccessors
+  void visitAccessorDecl(      AccessorDecl *d, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(d); }
+                          
+ void visitBreakStmt(             BreakStmt *s, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(s); }
+ void visitContinueStmt(       ContinueStmt *s, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(s); }
+ void visitFallthroughStmt( FallthroughStmt *s, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(s); }
+ void visitFailStmt(               FailStmt *s, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(s); }
+ void visitThrowStmt(             ThrowStmt *s, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(s); }
+ void visitPoundAssertStmt( PoundAssertStmt *s, ASTScopeImpl *p, DeferredNodes&) { p->widenSourceRangeForIgnoredASTNode(s); }
+
+    // clang-format on
+                          
+
+#pragma mark simple creation ignoring deferred nodes
+// clang-format off
+  void visitSubscriptDecl(SubscriptDecl *e, ASTScopeImpl *p, DeferredNodes &) { p->createSubtree<SubscriptDeclScope>(e); }
+
+  void visitIfStmt(                   IfStmt *e, ASTScopeImpl *p, DeferredNodes &) { p->createSubtree<     IfStmtScope>(e); }
```

Comment:
Same here. It would be much easier to read and wouldn't have special formatting if you use a `#define` like I suggested above.

Reply:
Yup, done!